app/addons/documents/changes/components.react.jsx

```diff
@@ -0,0 +1,244 @@
+    },
+
+    componentWillUnmount: function () {
+      Stores.changesHeaderStore.off('change', this.onChange);
+    },
+
+    toggleFilterSection: function () {
+      Actions.toggleTabVisibility();
+    },
+
+    render: function () {
+      var tabContent = '';
+      if (this.state.showTabContent) {
+        tabContent = <ChangesFilter key="changesFilterSection" />;
+      }
+
+      return (
+        <div className="changes-header-section">
+          <ChangesHeaderTab onToggle={this.toggleFilterSection} />
+          <ReactCSSTransitionGroup transitionName="toggleChangesFilter" component="div" className="changes-tab-content">
+            {tabContent}
+          </ReactCSSTransitionGroup>
+        </div>
+      );
+    }
+  });
+
+
+  var ChangesHeaderTab = React.createClass({
+    propTypes: {
+      onToggle: React.PropTypes.func.isRequired
+    },
+
+    render: function () {
+      return (
+        <div className="dashboard-upper-menu">
+          <ul className="nav nav-tabs" id="db-views-tabs-nav">
+            <li>
+              <a href="#filter" onClick={this.props.onToggle} data-bypass="true" data-toggle="tab">
+                <i className="fonticon fonticon-plus"></i> Filter
+              </a>
+            </li>
+          </ul>
+        </div>
+      );
+    }
+  });
+
+
+  var ChangesFilter = React.createClass({
+    getStoreState: function () {
+      return {
+        filters: Stores.changesFilterStore.getFilters()
+      };
+    },
+
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+
+    componentDidMount: function () {
+      Stores.changesFilterStore.on('change', this.onChange, this);
+    },
+
+    componentWillUnmount: function () {
+      Stores.changesFilterStore.off('change', this.onChange);
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    getFilters: function () {
+      return _.map(this.state.filters, function (filter) {
+        return <Filter key={filter} label={filter} />;
+      }, this);
+    },
+
+    render: function () {
+      var filters = this.getFilters();
+
+      return (
+        <div className="tab-content">
+          <div className="tab-pane active" ref="filterTab">
+            <div className="changes-header js-filter">
+              <AddFilterForm tooltip={this.props.tooltip} filter={this.state.filter} onSubmit={this.submitForm} />
+              <ul className="filter-list">{filters}</ul>
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+
+  var AddFilterForm = React.createClass({
+    getInitialState: function () {
+      return {
+        filter: ''
+      };
+    },
+
+    getDefaultProps: function () {
+      return {
+        tooltip: ''
+      };
+    },
+
+    submitForm: function (e) {
+      e.preventDefault();
+      e.stopPropagation();
+      if (_.isEmpty(this.state.filter)) {
+        return;
+      }
+      Actions.addFilter(this.state.filter);
```

Comment:
This should also be a dumb component that just calls a callback on a submit form and not an action directly.